components/ui/card-collection/card-collection-types.tsx

```diff
@@ -0,0 +1,117 @@
+export enum CardType {
+  FIXED = 'FIXED',
+  EXPANDABLE = 'EXPANDABLE',
+  FILL_REMAINING_SPACE = 'FILL_REMAINING_SPACE'
+}
+
+export interface FixedCardInputData {
+  type: CardType.FIXED,
+  order: number,
+  columnCount: number,
+  cardId: string,
+  renderCardContent: (input: {cardConfigurations: CardConfiguration[]}) => React.ReactNode,
+  cardClassname: string,
+}
+
+export interface ExpandableCardInputData {
+  type: CardType.EXPANDABLE,
+  order: number,
+  expandedColumnCount: number,
+  isExpandedByDefault: boolean,
+  cardId: string,
+  renderCardContent: (input: {cardConfigurations: CardConfiguration[]}) => React.ReactNode,
+  cardClassname: string,
+  onCardSizeChange?: () => void,
+}
+
+export interface FillRemainingSpaceCardInputData {
+  type: CardType.FILL_REMAINING_SPACE,
+  order: number,
+  cardId: string,
+  renderCardContent: (input: {cardConfigurations: CardConfiguration[]}) => React.ReactNode,
+  cardClassname: string,
+  onCardSizeChange?: () => void,
+}
```

Comment:
These should inherit from a common interface, to avoid repeated code

```{typescript}
export interface CardInputDataDefault {
  type: CardType,
  order: number,
  cardId: string,
  renderCardContent: (input: {cardConfigurations: CardConfiguration[]}) => React.ReactNode,
  cardClassname: string,
}
```

Then we can reduce a lot of these repeated attributes, and the difference between the different types can be obvious.

Reply:
This is a great idea, done :)